docs/saltstack.md

```diff
@@ -51,3 +51,20 @@ Example:
 
 {% endmacro %}
 ```
+
+### Issue: Grains are not getting refreshed within the same salt state
+Reason: To apply the state, Salt needs to convert it from YAML format to a data structure. Since it is impossible to parse a YAML file that has Jinja statements inside, Salt needs to parse Jinja first.
```

Comment:
@mazinamdar please put a reference here to an original salt issue where good discussion happened

Reply:
comment is addressed